spec/unit/util/xendomains_spec.rb

```diff
@@ -7,60 +7,105 @@
 
   let(:xen0_domains) { my_fixture_read("xendomains") }
 
-  describe "when the xl command is present" do
```

Comment:
With the `Facter::Util::Xendomains.xen_command` method updated as described above, these tests become much more manageable. See https://github.com/whopper/facter/blob/test/xendomains/spec/unit/util/xendomains_spec.rb for a working example.

One of the issues you were running into here was that the call to `File.file?` to look for `xen-toolstack` was not being stubbed here, resulting in your new logic being skipped while running through the tests.